subworkflows/local/map.nf

```diff
@@ -20,39 +20,35 @@ workflow MAP {
     ch_versions       = Channel.empty()
     ch_multiqc_files  = Channel.empty()
 
-    if ( params.shard_fastq ) {
+    if ( params.run_fastq_sharding ) {
 
         ch_input_for_sharding = reads
 
         SEQKIT_SPLIT2( ch_input_for_sharding )
         ch_versions        = ch_versions.mix ( SEQKIT_SPLIT2.out.versions.first() )
 
-        ch_input_for_mapping = SEQKIT_SPLIT2.out.reads
+        reads = SEQKIT_SPLIT2.out.reads
             .transpose()
-            .combine(index)
-            .multiMap {
-                meta, reads, meta2, index ->
+            .map {
+                meta, reads ->
                     new_meta = meta.clone()
-                    new_meta.shard_number = reads.getName().replaceAll(/.*(part_\d+).fastq.gz/, '$1')
-                    new_meta.reference = meta2.id
-                    reads: [ new_meta, reads ] 
-                    index: [ meta2, index ]
-            } 
-
-    } else {
-
-        ch_input_for_mapping = reads
-                            .combine(index)
-                            .multiMap {
-                                meta, reads, meta2, index ->
-                                    new_meta = meta.clone()
-                                    new_meta.reference = meta2.id
-                                    reads: [ new_meta, reads ]
-                                    index: [ meta2, index]
-                            }
+                    new_meta.shard_number = reads.getName().replaceAll(/.*(part_\d+).(?:fastq|fq).gz/, '$1')
+                    [ new_meta, reads ] 
+            }
+            .groupTuple()
 
     }
 
+    ch_input_for_mapping = reads
+        .combine(index)
+        .multiMap {
+            meta, reads, meta2, index ->
+                new_meta = meta.clone()
+                new_meta.reference = meta2.id  
+                reads: [ new_meta, reads ] 
+                index: [ meta2, index ]
+        }
+
```

Comment:
Please don't overwrite the subworkflow input channel variable (`reads`). 
Instead go back to assigning `ch_input_for_mapping` on either clause of the if else.
The asynchronous nature of nextflow can make overwriting a headache to debug in places, and break `resume` at times. Even if it would not cause issues in this case, within a subworkflow that runs mostly linearly, I'd rather keep things consistent.